src/Api/Services/Notifications/NotificationsApiProcessor.php

```diff
@@ -3,7 +3,41 @@
 namespace App\Api\Services\Notifications;
 
 use App\Api\Services\Base\AbstractApiProcessor;
+use App\DB\Entity\User\User;
+use App\DB\EntityRepository\User\Notification\NotificationRepository;
+use App\Project\Remix\RemixManager;
+use App\User\Notification\NotificationManager;
+use Exception;
 
 final class NotificationsApiProcessor extends AbstractApiProcessor
 {
+  private NotificationRepository $notification_repository;
+  private NotificationManager $notification_manager;
+  private RemixManager $remix_manager;
+
+  public function __construct(
+        NotificationRepository $notification_repository,
+        NotificationManager $notification_manager,
+        RemixManager $remix_manager
+    ) {
+    $this->notification_repository = $notification_repository;
+    $this->notification_manager = $notification_manager;
+    $this->remix_manager = $remix_manager;
+  }
+
+  /**
+   * @throws Exception
+   */
+  public function markAllAsSeen(User $user): void
+  {
+    $notifications = $this->notification_repository->findBy(['user' => $user]);
```

Comment:
I know it already was coded that way, however, why would we need to fetch all notifications one by one to check their status and change it. A simple update query for all unseen notifications of the user would be enough, no need for any PHP logic. At least one should query only the unseen notifications of the user...


if you got some time I would highly appreciate if you could fix this performance issue. :)